binders/repeat.js

```diff
@@ -125,6 +130,11 @@ module.exports = function() {
         var nextSibling = previousView ? previousView.lastViewNode.nextSibling : this.element.nextSibling;
         this.element.parentNode.insertBefore(fragment, nextSibling);
       }, this);
+
+      // Expose the index
+      this.views.forEach(function(view, i) {
+        view.index = i;
+      });
```

Comment:
We shouldn't hard-code `index`. We should use `this.keyName`. And it needs to be added to the context rather than the view. `view.context[this.keyName] = i`